Assets/MixedRealityToolkit.SDK/Experimental/Features/Utilities/Migration/BoundsControlMigrationHandler.cs

```diff
@@ -0,0 +1,301 @@
+                    // create a minmaxscaleconstraint in case there's a min max scale set up
+                    if (boundingBox.ScaleMinimum != 0.0f || boundingBox.ScaleMaximum != 0.0f)
+                    {
+                        MinMaxScaleConstraint scaleConstraint = gameObject.AddComponent<MinMaxScaleConstraint>();
+                        scaleConstraint.ScaleMinimum = boundingBox.ScaleMinimum;
+                        scaleConstraint.ScaleMaximum = boundingBox.ScaleMaximum;
+                    }
+#pragma warning restore 0618
+                }
+
+                // migrate visuals
+                boundsControl.DrawTetherWhenManipulating = boundingBox.DrawTetherWhenManipulating;
+                boundsControl.HandlesIgnoreCollider = boundingBox.HandlesIgnoreCollider;
+                boundsControl.FlattenAxis = MigrateFlattenAxis(boundingBox.FlattenAxis);
+                boundsControl.BoxPadding = boundingBox.BoxPadding;
+                string configDir = GetBoundsControlConfigDirectory(boundingBox);
+                MigrateBoxDisplay(boundsControl, boundingBox, configDir);
+                MigrateLinks(boundsControl, boundingBox, configDir);
+                MigrateScaleHandles(boundsControl, boundingBox, configDir);
+                MigrateRotationHandles(boundsControl, boundingBox, configDir);
+                MigrateProximityEffect(boundsControl, boundingBox, configDir);
+
+                // debug properties
+                boundsControl.DebugText = boundingBox.debugText;
+                boundsControl.HideElementsInInspector = boundingBox.HideElementsInInspector;
+
+                // events
+                boundsControl.RotateStarted = boundingBox.RotateStarted;
+                boundsControl.RotateStopped = boundingBox.RotateStopped;
+                boundsControl.ScaleStarted = boundingBox.ScaleStarted;
+                boundsControl.ScaleStopped = boundingBox.ScaleStopped;
+
+                // destroy obsolete component
+                Object.DestroyImmediate(boundingBox);
+            }
+
+            // look in the scene for app bars and upgrade them too to point to the new component
+            MigrateAppBar(boundingBox, boundsControl);
+
+        }
+
+        virtual protected string GetBoundsControlConfigDirectory(BoundingBox boundingBox)
+        {
+            var scene = boundingBox.gameObject.scene;
+            if (scene != null)
+            {
+                string scenePath = scene.path;             
+                string sceneDir = System.IO.Path.GetDirectoryName(scenePath);
+                // if empty we're creating the folder in the asset root.
+                // This should only happen if we're trying to migrate a dynamically created
+                // gameobject - which is usually only in test scenarios
+                if (sceneDir == "")
+                {
+                    sceneDir = "Assets";
+                }
+
+                const string configDir = "BoundsControlConfigs";
```

Comment:
Interesting, why do we need to have a directory for holding configuration files for the bounds control? Surprised that bounds control needs separate files

Reply:
that comes from the scriptable object change - there was a request a while ago to make bounds control configurations into scriptable objects to make them shareable, similar than the interactable themes. So the configurations we're creating when migrating are the shareable scriptable objects you can plug into bounds control. When doing the automatic migration i have to create individual configs because i don't know which properties the users might have changed.